src/ProgressOnderwijsUtils/Html/HtmlAttribute.cs

```diff
@@ -75,6 +76,59 @@ public int Count
             }
         }
 
+        public bool HasClass(string className)
+        {
+            var classChars = className.AsSpan();
+            foreach (var attr in this) {
+                if (attr.Name == "class") {
+                    var haystack = attr.Value.AsSpan();
+                    while (haystack.Length > 0) {
+                        var endIdx = haystack.IndexOf(' ');
+
+                        ReadOnlySpan<char> head;
+                        if (endIdx == -1) {
+                            head = haystack;
+                            haystack = default;
+                        } else {
+                            head = haystack.Slice(0, endIdx);
+                            haystack = haystack.Slice(endIdx + 1);
+                        }
+                        if (head.SequenceEqual(classChars)) {
+                            return true;
+                        }
+                    }
+                }
+            }

```

Comment:
Ik zie hier wat overlap met de method Classes hieronder. Is het niet sneller/simpeler om hier Classes op te halen kijken of className daarin zit?

Reply:
Het is zeker niet sneller, want Classes doet een boel extra werk in de vorm van al die string-allocations en de array allocation; HasClass doet 0 allocations.

Reply:
Ik stop deze uitleg wel in een commentaar.